src/interactive/AccordionPanelGroup.jsx

```diff
@@ -19,155 +32,70 @@ class AccordionPanelGroup extends React.Component {
 			const nextPanelProps = panel.props;
-			const currentPanelProps = currentPanels[index].props;
-
-			Object.keys(nextPanelProps).forEach(key => {
-				// Don't waste time if we already know at least one prop
-				// on any panel has changed
-				if (panelPropsHaveChanged) {
-					return;
-				}
-
-				if (
-					isPrimitive(nextPanelProps[key]) &&
-					nextPanelProps[key] !== currentPanelProps[key]
-				) {
-					panelPropsHaveChanged = true;
-				}
-			});
-		});
-
-		if (panelPropsHaveChanged) {
-			this.accordionPanels = nextProps.accordionPanels.map(
-				this.initPanel,
-				this
-			);
-			const panelStates = this.getPanelStates();
-			this.setState({ panelStates });
-		}
-	}
 
-	/**
-	 * Get a mapping of panels and their open/closed state
-	 * @return {object} Panel open state keyed by panel id
-	 */
-	getPanelStates() {
-		return this.accordionPanels.reduce(function(stateObj, panel) {
-			stateObj[panel.props.clickId] = panel.props.isOpen;
-			return stateObj;
-		}, {});
-	}
-
-	/**
-	 * @description callback called when individual panel is clicked
-	 * and keeps track of open states in order to support AccordionPanelGroups that only
-	 * have one panel open at a time.
-	 * @param {Integer} clickedPanelId the clickId prop of the `AccordionPanel` component
-	 * @param {Boolean} isOpen whether to open the panel or not (!props.isOpen of the panel)
-	 * @returns {undefined}
-	 */
-	setPanelStates(clickedPanelId, isOpen) {
-		const panelStates =
-			!this.props.multiSelectable && isOpen
-				? Object.keys(this.state.panelStates).reduce((stateObj, clickId) => {
-						stateObj[clickId] = parseInt(clickId) === clickedPanelId;
-						return stateObj;
-					}, {})
-				: {
-						...this.state.panelStates,
-						[clickedPanelId]: isOpen,
-					};
-		this.setState({ panelStates });
-	}
+			if (nextPanelProps['isOpen'] !== prevState.panelStatesList[i]) {
+				panelPropsHaveChanged = true;
+				derivedPanelListState = getNewPanelState(
+					nextProps.multiSelectable,
+					prevState.panelStatesList,
+					{ panelIndex: i, isOpen: nextPanelProps['isOpen'] }
+				);
+			}
+		});
 
-	/**
-	 * @description inits a panel with props from the group and individual panel,
-	 * which is then stored as part of accordionPanels array
-	 * @param {Object} accordionPanel - `AccordionPanel` components to clone
-	 * @param {number} i - index of the `AccordionPanel` used as the key
-	 * @returns {Component} `AccordionPanel` component with props from `AccordionPanelGroup`
-	 */
-	initPanel(panel, index) {
-		const isOpen = panel.props.isOpen || false,
-			clickId = index;
-
-		const panelProps = {
-			indicatorAlign: this.props.indicatorAlign,
-			indicatorIcon: this.props.indicatorIcon,
-			indicatorIconActive: this.props.indicatorIconActive,
-			indicatorSwitch: this.props.indicatorSwitch,
-			isOpen,
-			clickId,
-			setClickedPanel: this.setPanelStates.bind(this),
+		return {
+			panelStatesList: panelPropsHaveChanged
```

Comment:
`derivedPanelListState` is initialized as `undefined`, so you could remove the extra `panelPropsHaveChanged` flag and just check to see if the derived panel state has been set:

```
return derivedPanelListState && derivedPanelListState || prevState.panelStatesList;
```